Task2/Task2/Task2/Services/ServicesByProduct.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Task2.Exceptions;
+using Task2.Products;
+
+namespace Task2.Services
+{
+    public static class ServicesByProduct
+    {
+        public static Bolt AdditionOfTwoProducts(Bolt leftProduct, Bolt rightProduct)
+        {
+            if (leftProduct.ProductName.Equals(rightProduct.ProductName, StringComparison.InvariantCultureIgnoreCase))
+            {
+                var obj = new Bolt()
+                {
+                    ProductName = leftProduct.ProductName,
+                    PurchaseCost = Math.Round((leftProduct.PurchaseCost * leftProduct.NumberOfUnits
+                                               + rightProduct.PurchaseCost * rightProduct.NumberOfUnits)
+                                              / (leftProduct.NumberOfUnits + rightProduct.NumberOfUnits), 2),
+                    Margin = Math.Round((leftProduct.Margin * leftProduct.NumberOfUnits
+                                         + rightProduct.Margin * rightProduct.NumberOfUnits)
+                                        / (leftProduct.NumberOfUnits + rightProduct.NumberOfUnits), 2),
+                    NumberOfUnits = leftProduct.NumberOfUnits + rightProduct.NumberOfUnits
+                };
+
+                return obj;
+            }
+            else
+            {
+                throw new ProductNameException("Products have different Product Name");
+            }
+        }
+
+        public static Brick AdditionOfTwoProducts(Brick leftProduct, Brick rightProduct)
```

Comment:
The same issue about arguments null check.

Reply:
fixed